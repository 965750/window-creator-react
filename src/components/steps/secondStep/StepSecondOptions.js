import React, { Component } from 'react'
import { connect } from 'react-redux'
import { changeWindowDivision } from '../../../store/actions/creatorActions'
import DivisionInput from './DivisionInput'

class StepSecondOptions extends Component {
    changeDivision = (value, id) => {
        this.props.changeWindowDivision(value, id)
    }

    render() {
        return (
            <div className={`pt-2`}>
                <div className={`mb-10`}>
                    <p className="mb-4 text-darkText border-b border-mercury">
                        Door division
                    </p>
                    <DivisionInput
                        id="rows"
                        inputValue={this.props.window.rows.length - 1}
                        changeDivision={this.changeDivision}
                    />
                    <DivisionInput
                        id="columns"
                        inputValue={this.props.window.columns.length - 1}
                        changeDivision={this.changeDivision}
                    />
                </div>
            </div>
        )
    }
}

const mapStateToProps = state => {
    return {
        window: state.creator.window,
    }
}

const mapDispatchToProps = dispatch => {
    return {
        changeWindowDivision: (value, id) =>
            dispatch(changeWindowDivision(value, id)),
    }
}

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(StepSecondOptions)
