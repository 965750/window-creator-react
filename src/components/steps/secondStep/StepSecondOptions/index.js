import { changeWindowDivision } from '../../../../store/actions/creatorActions'
import { connect } from 'react-redux'
import { FormattedMessage } from 'react-intl'

import DivisionInput from '../DivisionInput/'
import PropTypes from 'prop-types'
import React, { Component } from 'react'

class StepSecondOptions extends Component {
  changeDivision = (value, id) => {
    this.props.changeWindowDivision(value, id)
  }

  render() {
    return (
      <div className="pt-2">
        <div className="mb-10">
          <p className="mb-4 text-osloGray border-b border-mercury">
            <FormattedMessage
              id="Door division"
              defaultMessage="Door division"
            />
          </p>
          <div className="flex justify-between">
            <p>
              <FormattedMessage
                id="Number of beams"
                defaultMessage="Number of beams"
              />
            </p>
            <DivisionInput
              id="rows"
              classes="mb-5"
              inputValue={this.props.window.rows.length - 1}
              changeDivision={this.changeDivision}
            />
          </div>
          <div className="flex justify-between">
            <p>
              <FormattedMessage
                id="Number of posts"
                defaultMessage="Number of posts"
              />
            </p>
            <DivisionInput
              id="columns"
              inputValue={this.props.window.columns.length - 1}
              changeDivision={this.changeDivision}
            />
          </div>
        </div>
      </div>
    )
  }
}

const mapDispatchToProps = (dispatch) => ({
  changeWindowDivision: (value, id) => dispatch(changeWindowDivision(value, id)),
})

StepSecondOptions.propTypes = {
  window: PropTypes.shape({
    color: PropTypes.string,
    width: PropTypes.number,
    height: PropTypes.number,
    doorType: PropTypes.number,
    rows: PropTypes.array,
    columns: PropTypes.array,
  }),
  changeWindowDivision: PropTypes.func,
}

export default connect(
  null,
  mapDispatchToProps,
)(StepSecondOptions)
