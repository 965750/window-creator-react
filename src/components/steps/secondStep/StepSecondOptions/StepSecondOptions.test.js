import '@testing-library/jest-dom/extend-expect'

import { applyMiddleware, compose, createStore } from 'redux';
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'
import { render, cleanup } from '@testing-library/react'

import StepSecondOptions from './index'
import messages from '../../../../messages-i18n'
import thunk from 'redux-thunk';
import React from 'react'

const initialStore = {
  theme: {
    lang: 'pl',
  },
}

const reducer = (state = initialStore, action) => {
  switch (action.type) {
    default:
      return state
  }
}

const store = createStore(
  reducer,
  compose(
    applyMiddleware(thunk),
  ),
)

const renderWithRedux = (component) => ({
  ...render(<Provider store={store}>{component}</Provider>),
})

describe('StepSecondOptions', () => {
  let wrapper
  const initialProps = {
    window: {
      color: '#000',
      width: 110,
      height: 230,
      doorType: 2,
      rows: [0, 1, 2],
      columns: [0, 1, 2],
    },
    changeWindowDivision: jest.fn(),
  }

  const setUp = (props = initialProps) => {
    wrapper = renderWithRedux(
      <IntlProvider
        locale="en"
        messages={messages[store.getState().theme.lang]}
      >
        <StepSecondOptions {...props} />
      </IntlProvider>,
    )
  }

  afterEach(cleanup)

  it('should match snapshot', () => {
    setUp()

    expect(wrapper.asFragment()).toMatchSnapshot()
  })
})
