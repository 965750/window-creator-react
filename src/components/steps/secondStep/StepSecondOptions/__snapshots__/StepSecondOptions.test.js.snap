// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepSecondOptions should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="pt-2"
  >
    <div
      class="mb-10"
    >
      <p
        class="mb-4 text-osloGray border-b border-mercury"
      >
        Podział drzwi
      </p>
      <div
        class="flex justify-between"
      >
        <p>
          Ilość belek
        </p>
        <div
          class="mb-5 flex"
          data-testid="divisionInputWrapper"
        >
          <div
            class="inputBox border border-gullGray text-center"
          >
            2
          </div>
          <button
            class="inputBox ml-1 bg-botticelli text-center cursor-pointer"
          >
            +
          </button>
          <button
            class="inputBox ml-1 bg-botticelli text-center cursor-pointer"
          >
            -
          </button>
        </div>
      </div>
      <div
        class="flex justify-between"
      >
        <p>
          Ilość słupków
        </p>
        <div
          class=" flex"
          data-testid="divisionInputWrapper"
        >
          <div
            class="inputBox border border-gullGray text-center"
          >
            2
          </div>
          <button
            class="inputBox ml-1 bg-botticelli text-center cursor-pointer"
          >
            +
          </button>
          <button
            class="inputBox ml-1 bg-botticelli text-center cursor-pointer"
          >
            -
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
